File: Samples/ResizeAppView/cppwinrt/Scenario2_Launch.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario2_Launch.h"
#include "Scenario2_Launch.g.cpp"

using namespace winrt;
using namespace Windows::Foundation;
using namespace Windows::UI::ViewManagement;
using namespace Windows::UI::Xaml;

namespace winrt::SDKTemplate::implementation
{
    Scenario2_Launch::Scenario2_Launch()
    {
        InitializeComponent();
        LaunchAtSize().IsChecked(ApplicationView::PreferredLaunchWindowingMode() != ApplicationViewWindowingMode::Auto);
    }

    // If you set the PreferredLaunchViewSize and PreferredLaunchWindowingMode
    // before calling Window.Current.Activate(), then it will take effect
    // on the current launch. Otherwise, it takes effect at the next launch.
    void Scenario2_Launch::LaunchAtSize_Click(IInspectable const&, RoutedEventArgs const&)
    {
        if (LaunchAtSize().IsChecked().Value())
        {
            // For best results, set the PreferredLaunchViewSize before setting
            // the PreferredLaunchWindowingMode.
            ApplicationView::PreferredLaunchViewSize({ 600, 500 });
            ApplicationView::PreferredLaunchWindowingMode(ApplicationViewWindowingMode::PreferredLaunchViewSize);
        }
        else
        {
            ApplicationView::PreferredLaunchWindowingMode(ApplicationViewWindowingMode::Auto);
        }
    }
}

File: archived/ResizeAppView/js/js/scenario2-launch.js
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";

    var ViewManagement = Windows.UI.ViewManagement;
    var ApplicationViewWindowingMode = ViewManagement.ApplicationViewWindowingMode;
    var ApplicationView = ViewManagement.ApplicationView;

    var page = WinJS.UI.Pages.define("/html/scenario2-launch.html", {
        ready: function (element, options) {
            launchAtSize.addEventListener("click", onLaunchAtSizeChanged);

            launchAtSize.checked = ApplicationView.preferredLaunchWindowingMode != ApplicationViewWindowingMode.auto;
        }
    });

    // If you set the preferredLaunchViewSize and preferredLaunchWindowingMode
    // before calling WinJS.UI.processAll, then it will take effect
    // on the current launch. Otherwise, it takes effect at the next launch.
    function onLaunchAtSizeChanged() {
        if (launchAtSize.checked) {
            // For best results, set the preferredLaunchViewSize before setting
            // the preferredLaunchWindowingMode.
            ApplicationView.preferredLaunchViewSize = { width: 600, height: 500 };
            ApplicationView.preferredLaunchWindowingMode = ApplicationViewWindowingMode.preferredLaunchViewSize;
        } else {
            ApplicationView.preferredLaunchWindowingMode = ApplicationViewWindowingMode.auto;
        }
    }
})();
